tezos/encoding/src/binary_writer.rs

```diff
@@ -472,6 +472,16 @@ fn find_value_in_record_values<'a>(
         .map(|(_, value)| value)
 }
 
+fn encoded_size_add(size: usize, val: usize) -> Result<usize, Error> {
+    size.checked_add(val)
+        .ok_or(Error::custom("Encoded message size overflow"))
+}
+
+fn encoded_size_sub(size: usize, val: usize) -> Result<usize, Error> {
+    size.checked_sub(val)
+        .ok_or(Error::custom("Encoded message size overflow"))
```

Comment:
@akoptelov 
could you please just change: overflow -> underflow?

Reply:
@bkontur From CPU point of view this is still overflow, see the std documentation: https://doc.rust-lang.org/std/primitive.usize.html#method.checked_sub

Reply:
@akoptelov 
aha, ok, ok, I just want to have different error messages for different cases - that we could differ directly from the log in case of error

Reply:
To my understanding this is very unlikely that this ever happen. And if it will, it will be quite hard reproducing it with backtrace option. So probably we can use different messages for each place instead. I'll do that.

Reply:
Please take a look if there is enough information now for possible troubleshooting.

Reply:
cool, looks ok :)